app/controllers/orders_controller.rb

```diff
@@ -37,10 +37,14 @@ def create
 
   def update
     @order = Order.find(params[:id])
-    if @order.update(order_params)
+    if @order.deliverer_id != nil
```

Comment:
fix style, move "redirect_to @order" to end of if/else statement & remove elsif statement